src/libraries/Native/Unix/System.Native/pal_time.c

```diff
@@ -53,34 +53,26 @@ int32_t SystemNative_UTimensat(const char* path, TimeSpec* times)
 // that is "nanoseconds per tick" in which case we need to scale before returning.
 uint64_t SystemNative_GetTimestampResolution()
 {
-#if HAVE_MACH_ABSOLUTE_TIME
-    mach_timebase_info_data_t mtid;
-
-    if (mach_timebase_info(&mtid) != KERN_SUCCESS)
+#if HAVE_CLOCK_GETTIME_NSEC_NP
+    if (clock_gettime_nsec_np(CLOCK_UPTIME_RAW) == 0)
     {
         return 0;
     }
-
-    // (numer / denom) gives you the nanoseconds per tick, so the below code
-    // computes the number of ticks per second. We explicitly do the multiplication
-    // first in order to help minimize the error that is produced by integer division.
-
-    return (SecondsToNanoSeconds * (uint64_t)(mtid.denom)) / (uint64_t)(mtid.numer);
-#else
+#endif
     // clock_gettime() returns a result in terms of nanoseconds rather than a count. This
```

Comment:
Since this always returns nanoseconds now, It may be best to delete this method and instead hardcode the conversions to nanoseconds on the managed side. It will be more efficient that way.